### ISSUE TITLE
Make hmt-escrow working in the Polygon Mainnet

### ISSUE THREAD

**Issue by vkomodey** [humanprotocol/hmt-escrow#307]
**Short Description**
Polygon has a different implementation of the HMToken. It has no bulk related methods. Because of this, `hmt-escrow` library can't be used in the Polygon, since `Escrow` contract uses `transferBulk` method

**Main Description**
Polygon Mainnet has a standard ERC20 interface and implementation for the HMToken. It is implemented via the standard bridge mechanism and represents a Proxy + Implementation
The Proxy - [0xc748B2A084F8eFc47E086ccdDD9b7e67aEb571BF](https://polygonscan.com/address/0x4948d86aa5f37fc07e3aa4a6f6f34f740b2752cc)
The Implementation - [0x4948d86aa5f37fc07e3aa4a6f6f34f740b2752cc](https://polygonscan.com/address/0x4948d86aa5f37fc07e3aa4a6f6f34f740b2752cc)

(Actually, the official type for the implementation is not a [ERC20](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/#methods), but [UChildERC20](https://polygonscan.com/address/0xfb935fbbacaa84a7b41e6ce05d62201bc75f68b7#code), which has methods intended for moving assets from Ethereum to Polygon(deposit and withdraw)

Due to this, hmt-escrow can't be used in the Polygon, since Escrow uses [transferBulk](https://github.com/humanprotocol/hmt-escrow/blob/master/contracts/Escrow.sol#L169) method during payouts

**Expected behavior**
[Job](https://github.com/humanprotocol/hmt-escrow/blob/master/hmt_escrow/job.py) should not fail in the polygon when it uses Escrow from the [contracts](https://github.com/humanprotocol/hmt-escrow/tree/master/contracts) folder(and particularly, when it  does bulk payouts)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200560623420491/1202088470037667) by [Unito](https://www.unito.io)

**Comment by e271828-** [comment]
@vkomodey this issue could be a bit clearer. What is the desired end behavior?

For example:

1. Add transferBulk equivalent to hmt-escrow Escrow contract.
2. Adjust hmt-escrow library to take `network` parameter in class setup, fall back to Escrow transferBulk method on networks where transferBulk is not natively available in the token contract.

**Comment by vkomodey** [comment]
@e271828- 
The list you provided looks more like list of solutions, but not a desired behavior. But, I agree that expected behavior description a little bit blurry, let me rephrase it

**Comment by vkomodey** [comment]
Details of the solution I propose might be viewed in the PR #295 

In a nutshell: the idea is to have separate contracts for separate networks(only when they differ, Ethereum Mainnet and Rinkeby are using same pack, Polygon using another one(updated HMT and Escrow)). Add a possibility to configure contracts you use via env variable or argument during setup

As an alternative I see updating Escrow `bulkPayOut` and do not include any other contracts. This is very easy to implement and will do the work now. But for the long term it might create ambiguity for developers, since they can't use all features of the HMT in the Polygon(and possibly in other future EVM networks)

**Comment by portuu3** [comment]
Solved with #295
